eplus.py

```diff
@@ -37,6 +44,48 @@ def _get_eplus_data(session, eplus_url):
     return result
 
 
+class EplusSessionUpdater(Thread):
+    def __init__(self, session, eplus_url):
+        self._eplus_url = eplus_url
+        self._session = session
+        self._closed = Event()
+
+        super().__init__(name='EplusSessionUpdater', daemon=True)
+
+    def close(self):
+        log.info('[EplusSessionUpdater] Closing...')
+        self._closed.set()
+
+    def run(self):
+        while True:
+            if self._closed.is_set():
+                return
+
+            # Create a new session, and send a request to Eplus url to obtain the cookies4
+            log.debug('[EplusSessionUpdater] Refreshing cookies...')
+            fresh_response = self._session.http.get(self._eplus_url, headers={
+                'Cookie': ''
+            })
```

Comment:
to follow up on my other comments, basically this `get` needs to come from a different HTTPSession (and not `self._session.http`). 

And then cookie updating that happens next would stay the same - using the original `self._session.http.cookies.update()`